### ISSUE TITLE
Sometimes the extension is not running

### ISSUE THREAD

**Issue by zhuoyw** [ToranSharma/Duo-Strength#79]
I am using Firefox (76.0.1). The extension is not working sometimes (after I refresh it). 

In the console, I can not see any errors from Duo-Strength. Any ideas?

![image](https://user-images.githubusercontent.com/5935782/83473356-39a0f980-a43e-11ea-978a-528fd7d91099.png)

**Comment by ToranSharma** [comment]
Hi, sorry for the slow reply, I haven't been getting email notifications for a over a week so I have only just seen this.

Is this still happening?

I will have a look into this and see if can see any issues.

Toran

**Comment by ToranSharma** [comment]
Accidentally clicked close issue.

**Comment by zhuoyw** [comment]
Yes. I have updated firefox and it is still happening.

**Comment by ToranSharma** [comment]
This is potentially related to issue #80, where the username is not able to be defined, and so the user's extended profile data cannot be requested. I have come up with fix for this that will be released in v1.3.13.

Would you be able to report back if the issue still persists after you have received this update. You can check your version number in at the top of the options popout window that is opened when you click the Duo Strength icon:
![image](https://user-images.githubusercontent.com/4559968/84287547-f595af00-ab37-11ea-8747-65d64811036d.png)

**Comment by zhuoyw** [comment]
The problem persists after the update. I think the extension is just not activated. I have no experience in brower debugging...

The extension works well in my Chrome though.

**Comment by ToranSharma** [comment]
Okay, it is interesting that it is a Firefox only issue.

When the extension is not being displayed, is the Duo Strength logo shown in the browser tool bar? 

![image](https://user-images.githubusercontent.com/4559968/84395765-0ad11300-abf6-11ea-8cb3-35d99c64dcf5.png)

If it is there, does clicking on it bring up the options popup window as normal?

**Comment by zhuoyw** [comment]
Yes. I can see the options. I also find that when I reinstall the extension, it always works the first time. 

If it works well in your firefox, maybe the problem is my configurations (other extensions)? I can't reset my browser. I think I will use this great extension only in chrome for now. 😂

**Comment by ToranSharma** [comment]
I think I may have just encountered this behaviour in my Firefox. As you have reported, there are no errors in the console log, and none of the Duo Strength features or working. It does indeed look like the script isn't being injected.

In trying to debug what is going on I am getting some strange errors in the inspection window for the background script of the extension.

I have not yet found a way to consistently reproduce the behaviour as of yet.

**Comment by ToranSharma** [comment]
I have done some playing around, and managed to reproduce the behaviour intermittently several times now. I have loaded the extension unpacked from source as a temporary extension and still get the behaviour sometimes too.

I have added in some logs at different times in the duoStrength.js content script to see where these are going wrong. from what I can see, the script does get injected and run every time. However when the extension doesn't work it is because the function that should be called on document.body.onload, doesn't get called.

![image](https://user-images.githubusercontent.com/4559968/84818249-20877380-b00e-11ea-97aa-0f7d28d924e5.png)

When everything works, the console log inside the body.onload funtion is printed:
![image](https://user-images.githubusercontent.com/4559968/84818445-693f2c80-b00e-11ea-91a7-cf60e2a5dc3f.png)

It seems like what is happening is that sometimes the document's ready state is past the point of running its onload event handler by the time the content script is injected into the page, and so it is not run. I don't know why this is inconstant, nor why it has only started happening recently.

In the develop branch I had already actually changed this to check the document ready state and act appropriately, in hopes of some increase in load speed. Trying out the develop branch with is change fixes the issue.

I will pull that change into the master branch and release the fix as v1.3.15
